base/pkg/read.jl

```diff
@@ -64,7 +64,11 @@ function isfixed(pkg::AbstractString, prepo::LibGit2.GitRepo, avail::Dict=availa
     LibGit2.isdirty(prepo) && return true
     LibGit2.isattached(prepo) && return true
     LibGit2.need_update(prepo)
-    LibGit2.iszero(LibGit2.revparseid(prepo, "HEAD:REQUIRE")) && isfile(pkg,"REQUIRE") && return true
+    try
+        LibGit2.revparseid(prepo, "HEAD:REQUIRE")
+    catch e
```

Comment:
if it's going to be a try catch (which it shouldn't) then it needs to be specific about what it's expecting